From c2829f3b1e29b4f6b058509e6c8a8d5fdccb2736 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 18 Sep 2025 15:51:37 +0200
Subject: [PATCH] board: solidrun: imx8mm_som: fix sd card-detect

When setting SD-Card pinmux from SPL also setup card-detect signal.

Fixes boot error below:

Trying to boot from MMC1
MMC: no card present
spl: mmc init failed with error: -123
SPL: failed to boot from all boot devices

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 board/solidrun/imx8mm_som/spl.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/board/solidrun/imx8mm_som/spl.c b/board/solidrun/imx8mm_som/spl.c
index d54b0bc2..03dea3cd 100644
--- a/board/solidrun/imx8mm_som/spl.c
+++ b/board/solidrun/imx8mm_som/spl.c
@@ -210,6 +210,7 @@ static iomux_v3_cfg_t const usdhc2_pads[] = {
 	IMX8MM_PAD_SD2_DATA2_USDHC2_DATA2 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
 	IMX8MM_PAD_SD2_DATA3_USDHC2_DATA3 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
 	IMX8MM_PAD_SD2_RESET_B_GPIO2_IO19 | MUX_PAD_CTRL(USDHC_GPIO_PAD_CTRL),
+	IMX8MM_PAD_SD2_CD_B_USDHC2_CD_B  | MUX_PAD_CTRL(0),
 };
 
 static struct fsl_esdhc_cfg usdhc_cfg[2] = {
-- 
2.51.0

